src/RadixDlt.NetworkGateway.GatewayApi/gateway-api-schema.yaml

```diff
@@ -3514,48 +3590,51 @@ components:
             An optional transaction message. Only affects the costing.
             This type is defined in the Core API as `TransactionMessage`. See the Core API documentation for more details.
         flags:
-          type: object
-          required:
-            - use_free_credit
-            - assume_all_signature_proofs
-            - skip_epoch_check
-          properties:
-            use_free_credit:
-              description: |
-                Whether to use a virtual, preview-only pool of XRD to pay for all execution fees.
-              type: boolean
-            assume_all_signature_proofs:
-              description: |
-                Whether the virtual signature proofs should be automatically placed in the auth zone.
-              type: boolean
-            skip_epoch_check:
-              description: |
-                Whether to skip the epoch range check (i.e. ignoring the `start_epoch_inclusive` and
-                `end_epoch_exclusive` parameters, if specified).
+          $ref: "#/components/schemas/TransactionPreviewFlags"
 
-                Note: effectively, without an epoch range, the Radix Engine cannot perform the
-                *intent hash duplicate* detection, which means that this check will be skipped as well.
-              type: boolean
-            disable_auth_checks:
-              description: |
-                Whether to skip the auth checks during execution.
-
-                This could be used to e.g.:
-                * Preview protocol update style transactions.
-                * Mint resources for previewing trades with resources you don't own. If doing this, be warned:
-                  * Only resources which were potentially mintable/burnable at creation time
-                    will be mintable/burnable, due to feature flags on the resource.
-                  * Please see the below warning about unexpected results if using this approach.
-
-                Warning: this mode of operation is quite a departure from normal operation:
-                * Calculated fees will likely be lower than a standard execution.
-                * This mode can subtly break invariants some dApp code might rely on, or result in unexpected
-                  behaviour, so the resulting execution result might not be valid for your needs. For example,
-                  if I used this flag to mint pool units to preview a redemption (or some dApp interaction which
-                  behind the scenes redeemed them), they'd redeem for less than they're currently worth,
-                  because the blueprint code relies on the total supply of the pool units to calculate their
-                  redemption worth, and you've just inflated the total supply through the mint operation.
-              type: boolean
+    TransactionPreviewFlags:
+      type: object
+      required:
+        - use_free_credit
```

Comment:
Could we align with the node on these fields and descriptions?

I've made all of these optional and added lots of detail to them.

Note that, to support compatibility with previous node versions, you will need to fill in the defaults for all flags -- simply set all flags to be false unless they're overriden in the user's request.

Reply:
ok fixed descriptions and made them all optional. no need to change anything in order to work with older nodes as our generator generates ctor with default values i.e:

``` public PreviewFlags(bool useFreeCredit = default(bool), bool assumeAllSignatureProofs = default(bool), bool skipEpochCheck = default(bool), bool disableAuthChecks = default(bool))
        ```

to make it clear in specs and in code i explicitly provided `default: false`  and it generated explicit defaults

```public PreviewFlags(bool useFreeCredit = false, bool assumeAllSignatureProofs = false, bool skipEpochCheck = false, bool disableAuthChecks = false)```